cupy/lib/polynomial.pyx

```diff
@@ -4,6 +4,37 @@ import cupy
 from cupy.core.core cimport ndarray
 
 
+def polyadd(a1, a2):
+    """Computes the sum of two polynomials.
+
+    Args:
+        a1 (cupy.ndarray or cupy.poly1d): first input polynomial.
+        a2 (cupy.ndarray or cupy.poly1d): second input polynomial.
+
+    Returns:
+        cupy.ndarray or cupy.poly1d: The sum of the inputs.
+
+    .. seealso:: :func:`numpy.polyadd`
+
+    """
+    truepoly = False
+    if isinstance(a1, poly1d):
+        a1 = a1.coeffs
+        truepoly = True
+    if isinstance(a2, poly1d):
+        a2 = a2.coeffs
+        truepoly = True
+    a1 = cupy.atleast_1d(a1)
+    a2 = cupy.atleast_1d(a2)
+    if a1.shape[0] < a2.shape[0]:
+        a1, a2 = a2, a1
+    a2 = cupy.pad(a2, (a1.shape[0] - a2.shape[0], 0))
+    val = a1 + a2
```

Comment:
"+=" fails with `TestPolyaddNdimShapeCombination` tests, `test_polyadd_diff_types_array` and `test_polyadd_diff_types_poly1d`

Reply:
If so how about custom `ElementwiseKernel` to reduce memory usage?

Reply:
`test_polyadd_diff_types_array` and  `test_polyadd_diff_types_poly1d` can pass using `result_type`  without needing a kernel. As for `TestPolyaddNdimShapeCombination` tests, I don't think a kernel will help. Please correct me if I'm wrong

Reply:
Please avoid allocating memory for padded `a2`.

Reply:
Please address this comment.

Reply:
I suggest removing the multidimensional inputs support ( **tests** to be precise) for 2 reasons:

1- Even if we guarantee the same behaviour as Numpy's, we won't be able to document it. Numpy states clearly that _Each input must be either a poly1d object or a 1D sequence of polynomial coefficients_ https://numpy.org/doc/stable/reference/generated/numpy.polyadd.html
2- It will solve the problem of memory allocation with no complex implementation. And `+=` can easily be used.

@asi1024 What do you think ?

Reply:
It's OK to remove multidimensional input support.